File: TheaterSuggestions/CSharp/ExternalDependencies/ReservationsProvider/ReservationsProvider.cs
using System.Collections.Generic;
using System.IO;

namespace ExternalDependencies.ReservationsProvider
{
    public class ReservationsProvider
    {
        private readonly Dictionary<string, ReservedSeatsDto> _repository = new Dictionary<string, ReservedSeatsDto>();

        public ReservationsProvider()
        {
            var directoryName = $"{GetExecutingAssemblyDirectoryFullPath()}\\AuditoriumLayouts\\";

            foreach (var fileFullName in Directory.EnumerateFiles($"{directoryName}"))

                if (fileFullName.Contains("_booked_seats.json"))
                {
                    var fileName = Path.GetFileName(fileFullName);

                    var eventId = Path.GetFileName(fileName.Split("-")[0]);

                    _repository[eventId] = JsonFile.ReadFromJsonFile<ReservedSeatsDto>(fileFullName);
                }
        }

        public ReservedSeatsDto GetReservedSeats(string showId)
        {
            if (_repository.ContainsKey(showId)) return _repository[showId];

            return new ReservedSeatsDto();
        }

        private static string GetExecutingAssemblyDirectoryFullPath()
        {
            var directoryName = System.IO.Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().CodeBase);

            if (directoryName.StartsWith(@"file:\"))
            {
                directoryName = directoryName.Substring(6);
            }

            return directoryName;
        }
    }
}

File: TheaterSuggestions/Java/ExternalApis/ExternalDependencies/src/main/java/com/baasie/ExternalDependencies/reservationsprovider/ReservationsProvider.java
package com.baasie.ExternalDependencies.reservationsprovider;

import com.baasie.ExternalDependencies.IProvideCurrentReservations;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.datatype.guava.GuavaModule;
import org.springframework.stereotype.Service;

import java.io.IOException;
import java.net.URISyntaxException;
import java.nio.file.DirectoryStream;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.HashMap;
import java.util.Map;

@Service
public class ReservationsProvider implements IProvideCurrentReservations {

    private Map<String, ReservedSeatsDto> repository = new HashMap<>();

    public ReservationsProvider() throws IOException, URISyntaxException {
        String jsonDirectory = Paths.get(ClassLoader.getSystemResource("AuditoriumLayouts").toURI()).toString();
        DirectoryStream<Path> directoryStream = Files.newDirectoryStream(Paths.get(jsonDirectory));

        for (Path path : directoryStream) {
            if (path.toString().contains("_booked_seats.json")) {
                String fileName = path.getFileName().toString();
                ObjectMapper mapper = new ObjectMapper().registerModule(new GuavaModule());
                repository.put(fileName.split("-")[0], mapper.readValue(path.toFile(), ReservedSeatsDto.class));
            }

        }
    }

    public ReservedSeatsDto getReservedSeats(String showId) {
        if (repository.containsKey(showId)) {
            return repository.get(showId);
        }
        return new ReservedSeatsDto();
    }
}